caffe2/operators/utility_ops.cu

```diff
@@ -31,7 +31,19 @@ bool SumOp<CUDAContext>::RunOnDevice() {
   } else if (Input(0).IsType<at::Half>()) {
     return DoRunWithType<at::Half, at::Half>();
   } else {
-    CAFFE_THROW("Unsupported inputs");
+    CAFFE_THROW("Unsupported inputs:", Input(0).dtype().name());
+  }
+  return false;
+}
+
+template <>
+bool SumGradientOp<CUDAContext>::RunOnDevice() {
+  CAFFE_ENFORCE_GT(InputSize(), 1);
+
+  if (Input(1).IsType<float>()) {
+    return DoRunWithType<float>();
+  } else {
```

Comment:
I think we need to support half in the gradient as well.

Reply:
OK ...